theano/tensor/tests/test_opt.py

```diff
@@ -3727,6 +3725,22 @@ def test_xor(self):
         self.assert_eqs_const(f, 0)
 
 
+    def test_stacktrace(self):
+        mode = theano.compile.get_default_mode().including(
+            'local_useless_elemwise_comparison')
+
+        x = T.vector('x', dtype=config.floatX)
+        f = theano.function([x], T.gt(x, x), mode=mode)
+        self.assertTrue(check_stack_trace(f, ops_to_check='last'))
+
+        f = theano.function([x], T.le(x, x), mode=mode)
+        self.assertTrue(check_stack_trace(f, ops_to_check='last'))
+
+        # Julian: I tried testing the stack trace for a bunch of different
+        # functions, including maximum and shapes, but other opts remove
+        # the stack traces in this case.
```

Comment:
For "minimum(x, x) => x" and "maximum(x, x) => x", the node itself is removed, so there is not much to check.
Same for "maximum(shape_i(...), 0) => shape_i(...)".
Indeed, no stack trace was copied in those cases.

However, it should work for:
"minimum(shape_i(...), 0) => 0"
"T.eq(shape_i(...), -X) => 0"